src/sonic-yang-models/yang-models/sonic-portchannel.yang

```diff
@@ -89,6 +94,33 @@ module sonic-portchannel {
 
 		} /* end of container PORTCHANNEL */
 
+
+		container PORTCHANNEL_INTERFACE {
+
+			description "PORTCHANNEL_INTERFACE part of config_db.json";
+
+			list PORTCHANNEL_INTERFACE_IPPREFIX_LIST {
+
+				key "name ip_prefix";
+
+				leaf name {
+					/* key elements are mandatory by default */
+					type leafref {
+						path "/lag:sonic-portchannel/lag:PORTCHANNEL/lag:PORTCHANNEL_LIST/lag:name";
+					}
+				}
+
+				leaf ip_prefix {
+					/* key elements are mandatory by default */
+					type union {
```

Comment:
@ArthiSivanantham 
feel free to declare a new type if you think, which is the union of these 2, say stypes:sonic-ip-prefix,

Reply:
Yes Praveen. As a good practice we can define sonic-ip-prefix (which is a union of sonic-ipv4-prefix and sonic-ipv6-prefix) and can use the new type here (as well as in sonic-vlan yang). But how does this solve the problem of IP prefix canonization if sonic-ipv4-prefix is exactly the same as ietf inet ipv4-prefix (and v6) ?

Reply:
Because the location of import will be /usr/local/share/yang/modules/ietf i.e. the IETF models installed by Libyang.

Reply:
Thanks for the explanation @praveen-li. I've addressed your comment.